packages/scripts/src/monitoring-dvm2.0/simulateVoting.ts

```diff
@@ -0,0 +1,663 @@
+// It is adapted from original DVM2.0 upgrade tests in packages/scripts/src/upgrade-tests/voting2/3_SimulateVoting.ts
+// intended to be run as regular health checks on the VotingV2 contract.
+// This script can be run on a mainnet or goerli fork, with the following command and environment variables:
+// CUSTOM_NODE_URL=https://<goerli OR mainnet>.infura.io/v3/<YOUR-INFURA-KEY> \
+// yarn hardhat run ./src/monitoring-dvm2.0/simulateVoting.ts
+
+import hre from "hardhat";
+import { strict as assert } from "assert";
+
+const { formatBytes32String, formatEther, parseEther, parseUnits, toUtf8Bytes } = hre.ethers.utils;
+
+import { BigNumber, BigNumberish, BytesLike, Signer } from "ethers";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+
+import { OptimisticOracleRequestStatesEnum, PriceRequestStatusEnum } from "@uma/common";
+import { OptimisticOracleV2Ethers, StoreEthers } from "@uma/contracts-node";
+
+import { REQUIRED_SIGNER_ADDRESSES, SECONDS_PER_DAY } from "../utils/constants";
+import { getContractInstance } from "../utils/contracts";
+import { getForkChainId, increaseEvmTime } from "../utils/utils";
+import {
+  commitVote,
+  createCommittedVote,
+  getUniqueVoters,
+  getVotingContracts,
+  revealVote,
+  unstakeFromStakedAccount,
+  VotingPriceRequest,
+} from "../utils/votingv2-utils";
+
+interface PriceRequestData {
+  originalAncillaryData: BytesLike;
+  proposedPrice: BigNumberish;
+  priceRequest: VotingPriceRequest;
+}
+interface RewardTrackers {
+  lastUpdateTimestamp: BigNumber;
+  staked: [BigNumber, BigNumber, BigNumber];
+  unclaimedRewards: [BigNumber, BigNumber, BigNumber];
+}
+
+const foundationAddress = REQUIRED_SIGNER_ADDRESSES.foundation;
+const accountWithUma = REQUIRED_SIGNER_ADDRESSES.account_with_uma;
+
+const zeroBigNumber = hre.ethers.BigNumber.from(0);
+
+// Constants hardcoded in the SlashingLibrary, needs to be updated here upon change.
+const wrongVoteSlashPerToken = parseEther("0.001");
+const noVoteSlashPerToken = parseEther("0.001");
+
+// Initial voter balances relative to GAT.
+const voter1RelativeGatFunding = parseEther("1.1"); // Voter 1 reaches GAT and SPAT.
+const voter2RelativeGatFunding = parseEther("0.55");
+const voter3RelativeGatFunding = parseEther("0.5");
+
+// Tested price identifier should be whitelisted.
+const priceIdentifier = formatBytes32String("NUMERICAL");
+
+// Ensure VotingV2 contract has at least this much total stake in Wei so that rewardsPerTokenStored do not overflow.
+const minimumStakedAmount = hre.ethers.BigNumber.from("1000");
+
+async function main() {
+  const rewardTrackers: RewardTrackers = {
+    lastUpdateTimestamp: zeroBigNumber,
+    staked: [zeroBigNumber, zeroBigNumber, zeroBigNumber],
+    unclaimedRewards: [zeroBigNumber, zeroBigNumber, zeroBigNumber],
+  };
+
+  // Initiates data request through Optimistic Oracle by requesting, proposing and disputing.
```

Comment:
seems a bit convoluted to do this via an OO request. Feels like we could easily do this more directly by calling onto the dvm from a registered contract from the finder and directly call `requestPrice` but I get that you might want this to be as close to a production style setup as possible.

Reply:
yes, the idea was to keep it as close to production as possible. When adapting this from original DVM2.0 upgrade test script I considered using OOv3 that should be easier to construct, but this can be changed as part of separate PR if we consider to simplify this.